tools/layers.py

```diff
@@ -95,3 +111,37 @@ def set_temporal_settings(
     tprops.setFixedDuration(time_step)
     tprops.setDurationUnits(unit)
     tprops.setIsActive(True)
+
+
+def evaluate_expressions(
+    exp: QgsExpression,
+    feature: Optional[QgsFeature] = None,
+    layer: Optional[QgsMapLayer] = None,
+    context_scopes: Optional[List[QgsExpressionContextScope]] = None,
+) -> Union[bool, int, str, float, None]:
+    """
+    Evaluate a QGIS expression
+    :param exp: QGIS expression
+    :param feature: Optional QgsFeature
+    :param layer: Optional QgsMapLayer
+    :param context_scopes: Optional list of QgsExpressionContextScopes
+    :return: evaluated value of the expression
+    """
+    context = QgsExpressionContext()
+    scopes = context_scopes if context_scopes else []
+
+    if layer:
+        # noinspection PyArgumentList
```

Comment:
Use #noqa instead?

Reply:
What's that PyArgumentList warning? Is it needed? If so it's better to use `#noqa Wxxx` if equivalent flake8 warning exists.

Reply:
This is not caught with flake8 but I removed it anyway :)